transforms/universal/pdf2md/python/src/pdf2md_transform.py

```diff
@@ -0,0 +1,346 @@
+                                    f"Exception {str(e)} processing file {archive_doc_filename}, skipping"
+                                )
+            
+            else:
+                logger.warning(
+                    f"File {file_name=} is not detected as PDF nor as ZIP but {kind=}. Skipping."
+                )
+
+            table = pa.Table.from_pylist(data)
+            metadata = {
+                "nrows": len(table),
+                "nsuccess": len(success_doc_id),
+                "nfail": len(failed_doc_id),
+                "nskip": len(skipped_doc_id),
+            }
+            return [
+                (TransformUtils.convert_arrow_to_binary(table=table), ".parquet")
+            ], metadata
+        except Exception as e:
+            print(f"Fatal error with file {file_name=}. No results produced.")
+            logger.error(f"Fatal error with file {file_name=}. No results produced.")
+            return [], {}
+
+# class PdfToMdRuntime(DefaultRayTransformRuntime):
+#     """
+#     Ingest to Parquet runtime support
+#     """
+
+#     def __init__(self, params: dict[str, Any]):
+#         """
+#         Create filter runtime
+#         :param params: parameters, that should include
+#             ingest_supported_langs_file_key: supported languages file
+#             ingest_detect_programming_lang_key: whether to detect programming language
+#             ingest_domain_key: domain
+#             ingest_snapshot_key: snapshot
+#         """
+#         super().__init__(params)
+
+#     def get_transform_config(
+#         self,
+#         data_access_factory: DataAccessFactoryBase,
+#         statistics: ActorHandle,
+#         files: list[str],
+#     ) -> dict[str, Any]:
+#         """
+#         Set environment for filter execution
+#         :param data_access_factory - data access factory
+#         :param statistics - reference to the statistics object
+#         :param files - list of files to process
+#         :return: dictionary of filter init params
+#         """
+
+#         # This cannot yet be done, because the converter is not easily serializable
+
+#         # self.models_dir = Path(self.params.get(ingest_modelsdir_key, "scratch"))
+#         # print("Downloading models")
+#         # logger.info("Downloading models")
+#         # DocumentConverter.download_models(
+#         #     artifacts_path=self.models_dir,
+#         #     artifacts_url=DocumentConverter.artifacts_urls["standard"],
+#         # )
+#         # print("Initializing converter")
+#         # logger.info("Initializing converter")
+#         # converter = DocumentConverter(artifacts_path=self.models_dir)
+#         # converter_refs = ray.put(converter)
+
+#         # logger.debug(f"Placed converter into Ray object storage under reference {converter_refs=}")
+#         # return {ingest_converter_refname: converter_refs} | self.params
+
+#         return self.params
+
+
+class PdfToMdTransformConfiguration(TransformConfiguration):
```

Comment:
Pdf2Md?

Reply:
We can easily rename all. At the moment we used the same scheme as in https://github.com/IBM/data-prep-kit/blob/dev/transforms/code/code2parquet/python/src/code2parquet_transform.py.
@daw3rd @blublinsky just let us know which naming convention is advised.

Reply:
has been changed